stdlib/REPL/src/LineEdit.jl

```diff
@@ -97,7 +97,7 @@ mutable struct PromptState <: ModeState
     # indentation of lines which do not include the prompt
     # if negative, the width of the prompt is used
     indent::Int
-    refresh_lock::Threads.AbstractLock
+    refresh_lock::Threads.SpinLock
```

Comment:
Of all the changes, this is the only one I worry about. Couldn't it be a `ReentrantLock` too?

Reply:
I only see this being created in one place and it is using a `SpinLock` there.

Reply:
Some packages might create their own Prompt? I'm not sure but maybe HeaderREPLs (which is kinda dead, but just as an example).

Reply:
This is a `PromptState` though and is created by `init_state`. I am not even sure you can hook into this.

Reply:
Sounds good then, carry on.